src/main/java/org/alfresco/dataprep/SitePagesService.java

```diff
@@ -552,21 +552,20 @@ public boolean createWiki(final String userName,
         body.put("pagecontent", content);
         body.put("tags", createTagsArray(tags));
         HttpResponse response = client.executeAndRelease(userName, password, body, put);
-        switch (response.getStatusLine().getStatusCode())
+
+        if(response.getStatusLine().getStatusCode() == HttpStatus.SC_OK)
         {
-            case HttpStatus.SC_OK:
-                if (logger.isTraceEnabled())
-                {
-                    logger.trace("Wiki page " + wikiTitle + " is created successfuly");
-                }
-                return true;
-            case HttpStatus.SC_NOT_FOUND:
-                throw new RuntimeException("Invalid site " + siteName);
-            default:
-                logger.error("Unable to create wiki page: " + response.toString());
-                break;
+            if (logger.isTraceEnabled())
+            {
+                logger.trace("Wiki page " + wikiTitle + " is created successfuly");
+            }
+            return true;
+        }
+        else
+        {
+            logger.error("Unable to create wiki page: " + response.toString());
+            throw new RuntimeException("Something went wrong while creating wiki page. Response status code: " + response.getStatusLine().getStatusCode());

```

Comment:
It would be nice to be able to easily extract the status code from the exception. e.g. You could create a custom RuntimeException that contains the status code.